### PR TITLE
Scheduler - A11y - Don't hide tooltip on item delete

### PR DESCRIPTION
**Pull request by Tucchhaa**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentTooltip.tests.js [hunk 1]
```diff
@@ -545,71 +544,6 @@ module('Integration: Appointment tooltip', moduleConfig, () => {
         assert.equal(scheduler.tooltip.getDateText(), 'February 9 11:00 AM - 12:00 PM', 'dates and time were displayed correctly');
     });
 
-    test('Click on tooltip-remove button should call scheduler.deleteAppointment and hide tooltip', async function(assert) {
```
Comment (Tucchhaa) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentTooltip.tests.js:548]: Removed this test, because there is a new jest test covering this case

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentTooltip.tests.js [hunk 2]
```diff
@@ -545,71 +544,6 @@ module('Integration: Appointment tooltip', moduleConfig, () => {
...
-                startDate: new Date(2015, 1, 9, 11, 0),
-                endDate: new Date(2015, 1, 9, 12, 0),
-                text: 'Task 2'
-            },
-            'processDeleteAppointment has a correct arguments');
-
-        assert.notOk(scheduler.tooltip.isVisible(), 'tooltip was hidden');
-    });
-
-    test('Click on tooltip-remove button should call scheduler.updateAppointment and hide tooltip, if recurrenceRuleExpr and recurrenceExceptionExpr is set', async function(assert) {
```
Comment (Tucchhaa) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentTooltip.tests.js:572]: Removed this test, because there is a new jest test covering this case

File: packages/devextreme/js/__internal/scheduler/appointments/m_appointment_collection.ts [hunk 3]
```diff
@@ -344,6 +360,36 @@ class SchedulerAppointments extends CollectionWidget<any> {
     });
   }
 
+  private hideTooltipIfNeed($element?: dxElementWrapper): void {
+    const { appointmentTooltip } = this.option();
+
+    if (appointmentTooltip.isShownForTarget($element)) {
+      appointmentTooltip.hide();
+    }
```
[the remark this answers is not in the document]
Comment (Tucchhaa) [reply packages/devextreme/js/__internal/scheduler/appointments/m_appointment_collection.ts]: when removing item it always has an element

File: packages/devextreme/js/__internal/scheduler/tooltip_strategies/m_tooltip_strategy_base.ts [hunk 4]
```diff
@@ -36,13 +39,25 @@ export class TooltipStrategyBase {
...
       this.extraOptions = extraOptions;
       this.showCore(target, dataList);
     }
   }
 
+  public setTarget($target: dxElementWrapper): void {
+    const originalAnimationValue = this.tooltip.option('animation');
+
+    this.tooltip.option('animation', null);
+    this.tooltip.option('target', $target);
```
[the remark this answers is not in the document]
Comment (Tucchhaa) [reply packages/devextreme/js/__internal/scheduler/tooltip_strategies/m_tooltip_strategy_base.ts]: done

File: packages/devextreme/js/__internal/scheduler/appointments/m_appointment_collection.ts [hunk 5]
```diff
@@ -323,19 +324,34 @@ class SchedulerAppointments extends CollectionWidget<any> {
         }
 
         if (item.needToRemove) {
-          item.element?.detach();
```
Comment (Tucchhaa) [inline packages/devextreme/js/__internal/scheduler/appointments/m_appointment_collection.ts:326]: no need for calling detach, can just remove element directly

File: packages/devextreme/js/__internal/scheduler/appointments/utils/get_view_model_diff.ts [hunk 6]
```diff
@@ -50,19 +59,39 @@ const isDataChanged = (
...
-  viewModelOld,
-  viewModelNext,
-  compareViewModel(appointmentDataSource),
-);
+): DiffItem<AppointmentViewModelPlain, AppointmentViewModelPlain>[] => {
+  const match = (
+    a: AppointmentViewModelPlain,
+    b: AppointmentViewModelPlain,
+  ): boolean => {
+    if ('items' in a && 'items' in b) {
```
Comment (sjbur) [inline packages/devextreme/js/__internal/scheduler/appointments/utils/get_view_model_diff.ts:76]: Found an edge case when the tooltip hides despite collector having appointments.
Paste this code and try to delete appointment "B"

```
$("#scheduler").dxScheduler({
                        dataSource: [
                            {
                                id: 1,
                                text: "A",
                                startDate: new Date(2025, 0, 8),
                                endDate: new Date(2025, 0, 11),
                            },
                            {
                                id: 2,
                                text: "B",
                                startDate: new Date(2025, 0, 8),
                                endDate: new Date(2025, 0, 11),
                            },
                            {
                                id: 3,
                                text: "C",
                                startDate: new Date(2025, 0, 8),
                                endDate: new Date(2025, 0, 10),
                            },
                            {
                                id: 4,
                                text: "D",
                                startDate: new Date(2025, 0, 8),
                                endDate: new Date(2025, 0, 9),
                            },
                        ],
                        views: [{ type: "month", maxAppointmentsPerCell: 1 }],
                        currentView: "month",
                        currentDate: new Date(2025, 0, 8),
                    });
```

I think the reason for this bug is the line here. We could make the equal function more strict

**Review by sjbur — CHANGES_REQUESTED** [review]

File: packages/devextreme/js/__internal/scheduler/__tests__/appointment_tooltip.test.ts [hunk 7]
```diff
@@ -46,200 +121,330 @@ describe('Appointment tooltip behavior', () => {
...
+        expect(onAppointmentDeleted).toHaveBeenCalledTimes(1);
+        expect(onAppointmentDeleted).toHaveBeenCalledWith(
+          expect.objectContaining({ appointmentData: expect.objectContaining({ text: 'Inside Collector Apt2' }) }),
+        );
       });
 
-      POM.getCollectorButton().click();
-      POM.tooltip.getDeleteButton(0).click();
-      POM.popup.deleteAppointmentButton.click();
+      it('should delete specific appointment on delete button click', async () => {
```
Comment (aleksei-semikozov) [inline packages/devextreme/js/__internal/scheduler/__tests__/appointment_tooltip.test.ts]: Test name says `delete button click` but body uses `pressDeleteKeyOnTooltipItem` (Delete key, not click)

```suggestion
      it('should delete specific appointment by Delete key', async () => {
```

Comment (Tucchhaa) [reply packages/devextreme/js/__internal/scheduler/__tests__/appointment_tooltip.test.ts]: applied

File: packages/devextreme/js/__internal/scheduler/appointments/utils/get_view_model_diff.ts (hunk 6 above)
Comment (Tucchhaa) [reply packages/devextreme/js/__internal/scheduler/appointments/utils/get_view_model_diff.ts:76]: Thanks, I have fixed this case

**Review by sjbur — APPROVED** [review]
